docs/SPEC.md

```diff
@@ -830,6 +830,86 @@ Examples:
     f()
     a
 
+#### Named types
+
+A named type can be created using a type assignment statement.
+
+    TypeAssignement   = "type" identifier "=" TypeExpression
+    TypeExpression    = identifier
+                      | TypeParameter
+                      | ObjectType
+                      | ArrayType
+                      | GeneratorType
+                      | FunctionType .
+    TypeParameter     = "'" identifier .
+    ObjectType        = "{" PropertyTypeList [";" ObjectUpperBound ] "}" .
+    ObjectUpperBound  = "any" | PropertyTypeList .
+    PropertyTypeList  = PropertyType [ "," PropertyType ] .
+    PropertyType      = identifier ":" TypeExpression
+                      | string_lit ":" TypeExpression .
+    ArrayType         = "[]" TypeExpression .
+    GeneratorType     = "[...]" TypeExpression .
+    FunctionType      = FunctionParameterTypeList "->" TypeExpression
+    ParameterTypeList = "(" ParameterType { "," ParameterType } ")" .
+    ParameterType     = identifier ":" [ pipe_receive_lit ] TypeExpression .
+
+Named types are in their own namespace.
+It is possible for a value and a type to have the same identifier.
+The following named types are built-in.
+
+    bool     // boolean
+    int      // integer
+    uint     // unsigned integer
+    float    // floating point number
+    duration // duration of time
+    time     // time
+    string   // utf-8 encoded string
+    regexp   // regular expression
+    type     // a type that itself describes a type
+
+
+When an object's upper bound is not specified, it is assumed to be equal to its lower bound.
+
+Parameters to function types define whether the parameter is a pipe forward parameter and whether the parameter has a default value.
+The `<-` indicates the parameter is the pipe forward parameter.
+
+Examples:
+ 
+    // alias the bool type
+    type boolean = bool
```

Comment:
Are named types aliases? To be more precise, is `boolean` the same type as `bool`? Can `boolean` be used wherever `bool` can and vice versa?

Reply:
Yes, they are equivalent in every way.